openpgp.d.ts

```diff
@@ -80,7 +79,8 @@ export interface PrimaryUser {
 
 type AlgorithmInfo = {
   algorithm: enums.publicKeyNames;
-  bits: number;
+  bits?: number;
+  curve?: EllipticCurveName;
```

Comment:
This type is not quite right, because either `bits` or `curve` are always set (but not both).
A union type can fix the issue, but it's probably a bit too verbose for now, since I don't think the current declaration creates any actual issues.

Reply:
The current declaration will mean that the consumer will have to typically bang the value as in `algo.bits!` if they need the number.

```ts
// now
if(algo.algorithm === rsa) {
  algo.bits // number | undefined
  algo.curve // EllipticCurveName | undefined
}

// union
if(algo.algorithm === rsa) {
  algo.bits // type is number, because TS can infer it if the union was written correctly
  algo.curve // compilation error!
}
```

Generally, it is ergonomic for the user of the API if types are written to behave like this. But it really depends. For example in this case, there is also a downside:

```ts
// now, without if
algo.bits // number | undefined

// union, without if
algo.bits // compilation error!

// union without if, need to cast it
(algo as RsaAlgorithmInfo).bits // number
```

For the AlgorithmInfo in particular, maybe it's not quite worth it. But other places may benefit from this.

Reply:
Yeah, I think this would be cool, but doesn't necessarily have to happen in this PR. Also, with the current `algorithm: rsaSign | rsaEncrypt | rsaEncryptSign` this type (and check) might be a little overly verbose, maybe it would be nice to have a `type: 'rsa'` property.